app/views/users/_form.html.slim

```diff
@@ -3,8 +3,9 @@
   .row
     .col-xs-12
       = f.input :email
-      - if current_user == user
+      - if current_user == user || policy(User).reset_password?
```

Comment:
If my assumption above is correct you can just do `if policy(User).reset_password?` here